python3/shipper/shipper.py

```diff
@@ -0,0 +1,220 @@
+import gzip
+import json
+import logging
+import sys
+import time
+import os
+import urllib
+import io
+
+# set logger
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+class MaxRetriesException(Exception):
+    pass
+
+
+class UnauthorizedAccessException(Exception):
+    pass
+
+
+class BadLogsException(Exception):
+    pass
+
+
+class UnknownURL(Exception):
+    pass
+
+
+class GzipLogRequest(object):
+
+    def __init__(self, max_size_in_bytes):
+        self._max_size_in_bytes = max_size_in_bytes
+        self._decompress_size = 0
+        self._logs_counter = 0
+        self._logs = io.BytesIO()
+        self._writer = gzip.GzipFile(mode='wb', fileobj=self._logs)
+        self._http_headers = {"Content-Encoding": "gzip", "Content-type": "application/json"}
+
+    def __len__(self):
+        return self._logs_counter
+
+    def bytes(self):
+        return bytes(self._logs.getvalue())
+
+    # GZIP
+    def write(self, log):
+        log = bytes(log, 'utf-8')
+        self._writer.write("\n" + log) if self._decompress_size else self._writer.write(log)
```

Comment:
You need to decide if to add \n before you turn the string to bytes otherwise you're adding a new line from type string to a log from type bytes. 

Like you did in the String class:)

Reply:
fixed, the newline is added in bytes